src/KubernetesClient/ModelExtensions.cs

```diff
@@ -0,0 +1,356 @@
+        }
+
+        /// <summary>Determines whether one object is owned by another.</summary>
+        public static bool IsOwnedBy(this IMetadata<V1ObjectMeta> obj, IKubernetesObject<V1ObjectMeta> owner) =>
+            FindOwnerReference(obj, owner) >= 0;
+
+        /// <summary>Gets the labels of a Kubernetes object.</summary>
+        public static IDictionary<string, string> Labels(this IMetadata<V1ObjectMeta> obj) => obj.Metadata?.Labels;
+
+        /// <summary>Gets the name of a Kubernetes object.</summary>
+        public static string Name(this IMetadata<V1ObjectMeta> obj) => obj.Metadata?.Name;
+
+        /// <summary>Gets the namespace of a Kubernetes object.</summary>
+        public static string Namespace(this IMetadata<V1ObjectMeta> obj) => obj.Metadata?.NamespaceProperty;
+
+        /// <summary>Gets the owner references of a Kubernetes object.</summary>
+        public static IList<V1OwnerReference> OwnerReferences(this IMetadata<V1ObjectMeta> obj) => obj.Metadata?.OwnerReferences;
+
+        /// <summary>Removes the given finalizer from a Kubernetes object if it exists.</summary>
+        /// <returns>Returns true if the finalizer was removed and false if it didn't exist.</returns>
+        public static bool RemoveFinalizer(this IMetadata<V1ObjectMeta> obj, string finalizer)
+        {
+            if (obj == null) throw new ArgumentNullException(nameof(obj));
+            if (string.IsNullOrEmpty(finalizer)) throw new ArgumentNullException(nameof(finalizer));
+            return obj.Finalizers() != null && obj.Metadata.Finalizers.Remove(finalizer);
+        }
+
+        /// <summary>Removes the first <see cref="V1OwnerReference"/> that matches the given object and returns it, or returns null if no
+        /// matching reference could be found.
+        /// </summary>
+        public static V1OwnerReference RemoveOwnerReference(this IMetadata<V1ObjectMeta> obj, IKubernetesObject<V1ObjectMeta> owner)
+        {
+            int index = FindOwnerReference(obj, owner);
+            V1OwnerReference ownerRef = index >= 0 ? obj.Metadata.OwnerReferences[index] : null;
+            if (index >= 0) obj.Metadata.OwnerReferences.RemoveAt(index);
+            return ownerRef;
+        }
+
+        /// <summary>Removes all <see cref="V1OwnerReference">owner references</see> that match the given predicate, and returns true if
+        /// any were removed.
+        /// </summary>
+        public static bool RemoveOwnerReferences(this IMetadata<V1ObjectMeta> obj, Predicate<V1OwnerReference> predicate)
+        {
+            if (obj == null) throw new ArgumentNullException(nameof(obj));
+            if (predicate == null) throw new ArgumentNullException(nameof(predicate));
+            bool removed = false;
+            IList<V1OwnerReference> refs = obj.Metadata?.OwnerReferences;
+            if (refs != null)
+            {
+                for (int i = refs.Count-1; i >= 0; i--)
```

Comment:
Use `IList<T>.RemoveAll(...)`?

https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.removeall?view=netcore-3.0

Reply:
That's List\<T\>.RemoveAll. IList\<T\> has no such method.